pipelined/bevy_pbr2/src/render/mod.rs

```diff
@@ -379,30 +536,107 @@ pub fn queue_meshes(
+                                },
+                                BindGroupEntry {
+                                    binding: 1,
+                                    resource: BindingResource::TextureView(
+                                        &base_color_texture_view,
+                                    ),
+                                },
+                                BindGroupEntry {
+                                    binding: 2,
+                                    resource: BindingResource::Sampler(&base_color_sampler),
+                                },
+                                BindGroupEntry {
+                                    binding: 3,
+                                    resource: BindingResource::TextureView(&emissive_texture_view),
+                                },
+                                BindGroupEntry {
+                                    binding: 4,
+                                    resource: BindingResource::Sampler(&emissive_sampler),
+                                },
+                                BindGroupEntry {
+                                    binding: 5,
+                                    resource: BindingResource::TextureView(
+                                        &metallic_roughness_texture_view,
+                                    ),
+                                },
+                                BindGroupEntry {
+                                    binding: 6,
+                                    resource: BindingResource::Sampler(&metallic_roughness_sampler),
+                                },
+                                BindGroupEntry {
+                                    binding: 7,
+                                    resource: BindingResource::TextureView(&occlusion_texture_view),
+                                },
+                                BindGroupEntry {
+                                    binding: 8,
+                                    resource: BindingResource::Sampler(&occlusion_sampler),
+                                },
+                            ],
+                            label: None,
+                            layout: &pbr_shaders.material_layout,
+                        })
+                    });
 
             mesh_meta.mesh_draw_info.push(MeshDrawInfo {
                 material_bind_group_key,
             });
 
+            // NOTE: row 2 of the view matrix dotted with column 3 of the model matrix
+            //       gives the z component of translation of the mesh in view space
+            let mesh_z = view_row_2.dot(mesh.transform.col(3));
+            // FIXME: Switch from usize to u64 for portability and use sort key encoding
+            //        similar to https://realtimecollisiondetection.net/blog/?p=86 as appropriate
+            // FIXME: What is the best way to map from view space z to a number of bits of unsigned integer?
```

Comment:
I think the real answer is that we shouldn't do that :)

If we make Drawable generic, we can make sort_key whatever type we want

Reply:
We just need to make sure that the type we choose for a given phase is suitable for all possible things in that phase

Reply:
Are we concerned about performance of not encoding into a directly comparable data type like usize or u64 or so? Depending on the sorting algorithm, a lot of comparisons could be made and if we’re doing something like A.a < B.a || A.b < B.b || … then it could result in a lot more comparison operations in practice. @mtsr raised this point to me earlier.

Reply:
I think its worth benchmarking when we make the switch.